### PR TITLE
Correlation.py update and multiple test cases [WIP]

### PR DESCRIPTION
**Pull request by beyucel**
- Correlation cut-off issue solved
- Nonperiodic two-point stats issue solved
- Multiple test cases added for different input shapes

I know it won't pass the tests right now (due to pylint etc.) I will clean up the code and update the PR. While I am doing that if you have any suggestions (like asking for additional test cases). please  let me know

### CODE REVIEW

**Comment by beyucel** [comment]
I guess conda built has some  package conflict issues

File: pymks/fmks/correlations.py [hunk 1]
```diff
@@ -164,20 +163,45 @@ def two_point_stats(arr1, arr2, periodic_boundary=True, cutoff=None):
...
+        cutoff = np.floor((np.min(arr1.shape[1:]) - 1) / 2)
+
+    nonperiodic_padder = sequence(
+        lambda x: np.pad(
+            x,
+            [(0, 0)] + [(cutoff, cutoff)] * (arr1.ndim - 1),
+            mode="constant",
+            constant_values=0,
+        ),
+        lambda x: x.rechunk(x.shape),
```
Comment (wd15) [inline pymks/fmks/correlations.py:175]: This loses the original chunk information that `x` had. We need to rechunk only the axes that are being padded not all the axes. The chunks need to stay the same, especially for the sample axis. I'm working on it. We also need tests for this.

Comment (beyucel) [reply pymks/fmks/correlations.py:175]: That makes sense! We should also consider 3D cases if we are rechunking fixed axes

**Comment by wd15** [comment]
It will work for any dimension. I'll add some 1D and 3D tests.

**Comment by wd15** [comment]
@beyucel, don't rebase this until I've given you my changes

**Comment by beyucel** [comment]
> @beyucel, don't rebase this until I've given you my changes

Sure

**Comment by wd15** [comment]
This doesn't look like it passed the tests!!! Can we unmerge this?

**Comment by wd15** [comment]
Why is this merged?

**Comment by wd15** [comment]
If this was merged by mistake then I'll try and I'll fix things.

**Comment by beyucel** [comment]
I was trying to solve that issue! I don't know how that happened. I just
merged your pr to my repo. Should I revert that or can you help me with
that?

**Comment by beyucel** [comment]
Sorry about that!

**Comment by wd15** [comment]
No worries. I'll fix it.

Don't revert though it's not that helpful

**Pull request merged**